### PR TITLE
Framework: Add docs to section-preload

### PR DESCRIPTION
**Pull request by blowery**
An attempt to capture why this weird thing exists. No functional changes, just docs.

### CODE REVIEW

File: client/sections-preload.js [hunk 1]
```diff
@@ -1,8 +1,35 @@
+/**
+ * sections-preload
+ *
+ * This is a simple eventbus that sections.js listens to to know when to preload sections.
+ * This exists because we needed a way to break the chunk dependency cycle. In days past,
+ * the preloader was part of sections.js, and to preload a module, you would import sections
+ * and call preload directly. Unfortunately, since all of the require.ensure calls live in sections,
+ * that make webpack think that every chunk that included sections was dependant on every other chunk.
```
Comment (lancewillett) [inline client/sections-preload.js]: s/make/makes/

File: client/sections-preload.js [hunk 2]
```diff
@@ -1,8 +1,35 @@
...
+ * This ballooned compile times and made module analysis in chunks very difficult.
+ *
+ * To break the dependency cycle, we introduced sections-preload, which had no dependencies. Now modules
+ * that want to preload a section can import sections-preload instead. sections also imports sections-preload
```
Comment (lancewillett) [inline client/sections-preload.js]: Capital s on `Sections also imports...`

**Comment by blowery** [comment]
@lancewillett reworded a bit. thoughts?

**Comment by lancewillett** [comment]
👍 Nice and concise.

**Pull request merged**